src/library_exceptions.js

```diff
@@ -95,46 +104,17 @@ var LibraryExceptions = {
 #endif
       return prev === 1;
     };
-  },
 
-  $CatchInfo__docs: `
-  /**
-   * @constructor
-   * @param {number=} ptr
-   */`,
-  $CatchInfo__deps: ['$ExceptionInfo', '__cxa_is_pointer_type'],
-  // This native structure is returned from __cxa_find_matching_catch, and serves as catching
-  // context, i.e. stores information required to proceed with a specific selected catch. It stores
-  // base and adjusted pointers of a thrown object. It is allocated dynamically and should be freed
-  // when it is done with a specific catch (i.e. either in __cxa_end_catch when caught or in
-  // __resumeException when no catch clause matched). The class itself is just a native pointer
-  // wrapper, and contains all the necessary accessors for the fields in the native structure.
-  // ptr - Native structure pointer to wrap, the structure is allocated when not specified.
-  $CatchInfo: function(ptr) {
-
-    this.free = function() {
-      _free(this.ptr);
-      this.ptr = 0;
-    };
-
-    this.set_base_ptr = function(basePtr) {
-      {{{ makeSetValue('this.ptr', '0', 'basePtr', '*') }}};
-    };
-
-    this.get_base_ptr = function() {
-      return {{{ makeGetValue('this.ptr', '0', '*') }}};
-    };
+    this.get_adjusted_ptr_addr = function() {
+      return this.ptr + {{{ C_STRUCTS.__cxa_exception.adjustedPtr }}};
```

Comment:
This returns the `ptr` + the `adjustedPtr.` Should the field then be called 'adjustment' as it is not a pointer itself?

The `set_` function does not do a subtraction of `ptr`, which is not symmetrical with this function. Perhaps one should be changed?

Reply:
No, this return ptr + the offset of adjustedPtr within the struct... so its finding the location where the adjusted pointer is stored.

Reply:
I inlined this function into is caller so its more obvious why/how its used.